reporting/test/request-reporter.test.js

```diff
@@ -0,0 +1,163 @@
+/* eslint-disable prettier/prettier */
+/**
+ * WhoTracks.Me
+ * https://ghostery.com/whotracksme
+ *
+ * Copyright 2017-present Ghostery GmbH. All rights reserved.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0
+ */
+
+import 'fake-indexeddb/auto';
+import { IDBFactory } from 'fake-indexeddb';
+import chrome from 'sinon-chrome';
+import sinon from 'sinon';
+import { expect } from 'chai';
+
+import { playScenario } from './helpers/scenariors.js';
+import { base64ToArrayBuffer } from './helpers/fetch-mock.js';
+
+import { setLogLevel } from '../src/logger.js';
+import RequestReporter from '../src/request-reporter.js';
+import WebRequestPipeline from '../src/webrequest-pipeline/index.js';
+
+const config = {
+  configUrl: 'config',
+  remoteWhitelistUrl: 'whitelist',
+  localWhitelistUrl: 'local',
+};
+
+describe('RequestReporter', function () {
+  before(function () {
+    setLogLevel('error');
+    chrome.storage.session = chrome.storage.local;
+    globalThis.chrome = chrome;
+    const oldFetch = globalThis.fetch;
+    sinon.stub(globalThis, 'fetch').callsFake((...args) => {
+      const url = args[0];
+      if (url.startsWith(config.configUrl)) {
+        return Promise.resolve({
+          ok: true,
+          async json() {
+            return {};
+          },
+        });
+      }
+
+      if (url.startsWith(config.remoteWhitelistUrl)) {
+        return Promise.resolve({
+          ok: true,
+          async json() {
+            return {
+              version: '2018-10-11',
+              useDiff: false,
+            };
+          },
+          async arrayBuffer() {
+            return base64ToArrayBuffer('AAAAAgrdwUcnN1113w==');
```

Comment:
Could be helpful to comment what this encodes.

Reply:
ok. for the future, I want to make a better utility to create actual bloom filters